### ISSUE TITLE
atclient_connection_send fails to send cram command to secondary server

### ISSUE THREAD

**Issue by srieteja** [atsign-foundation/at_c#426]
### Describe the bug

I have been trying to send the cram command to the secondary server. The mbedtls_ssl_write (used by atclient_connection_send) somehow returns a valid int value but the cram command is not received on the server side. Then the program exits with 'mbedtls_ssl_read failed with exit code -26624' (26624 converted to hex is 0x6800).

Note that the rest of the commands before cram - fetching secondary address from root server and fetching challenge from secondary(from:@atsign) are executing as expected.

### Steps to reproduce

1. First I 'fetch the challenge from the server'
2. Then I 'generate a digest with this challenge and cram secret'
3. And then 'send the command _cram:digest/r/n'_
4. This command does not reach the server and also the existing socket is terminated

### Expected behavior

The command should go through to the server and the server should respond with 'data:success'

### Screenshots

Note: The server is using my local time but the client seems to be using UTC

Server:
<img width="1440" alt="Screenshot 2024-10-15 at 5 08 28 AM" src="https://github.com/user-attachments/assets/df2e8635-3e22-4e0f-90bb-74bc8dabd336">

Client:
<img width="1407" alt="Screenshot 2024-10-15 at 5 09 04 AM" src="https://github.com/user-attachments/assets/a95e695d-f0bc-44d8-ae2e-657b9ea7079e">

**Comment by JeremyTubongbanua** [comment]
Are you remembering to add `\r\n` at the end of the src buffer?
Also make sure not to send the null-terminator.

So sending `cram:$digest\r\n` is right, but `cram:$digest\r\n\0` might give you that error

**Comment by srieteja** [comment]
You are right @JeremyTubongbanua. The '\0' was causing the issue. Explanation: The fixed command length is 135 bytes (cram: + digest + \r + \n). When I allocated 135 bytes for the command and used snprintf to build the command the '\n' was being sacrificed to add the '\0' at the end. And when I allocated 136 bytes, the '\n' was intact but '\0' was being added at the end. This behaviour led me to believe that even when '\n' was part of the command this was not being sent to the server. Currently I am manually replacing the trailing '\0' with '\n' and this has solved this issue. Thank you @JeremyTubongbanua.
